src/components/App/App.jsx

```diff
@@ -142,21 +136,25 @@ const handleOnProductFormChange = (event) => {
 }
 
 // onclick function for each recipe card
-const handleRecipeCardClick = (showStatus, recipeId) => {
+const handleRecipeCardClick = async (showStatus, recipeId) => {
+  await handleGetRecipeInstructions(recipeId)
   handleRecipeModal(showStatus)
-  handleGetRecipeInstructions(recipeId)
 }
 
 // pop up recipe modal for specific recipe card
 const handleRecipeModal = (showStatus) => {
   setRecipeModalShow(showStatus)
 }
 
-// extract recipe instructions for recipe modal
-const handleGetRecipeInstructions = (recipeId) => {
-  //TODO: extract recipe instructions for specific given recipe using API
-  setRecipeInstructions(sampleRecipeInstructions)
-  setRecipeInfo(sampleRecipeInfo)
+// extract recipe info from backend for recipe modal
+const handleGetRecipeInstructions = async (recipeId) => {
+  try {
+    var { data } = await axios(recipeInfoUrl + recipeId)
+    setRecipeInfo(data)
+  } catch (error) {
+    console.log("error in fetching from backend")
```

Comment:
Should you also display some kind of error to the user? What is the experience for the user when this happens?